### ISSUE TITLE
`tools/autograd/derivatives.yaml` doesn't support methods on optional tensor

### ISSUE THREAD

**Issue by jjsjann123** [pytorch/pytorch#62934]
## 🐛 Bug

We should be able to access attributes from optional tensor in derivatives.yaml.

e.g.
```
- name: fwd_func(Tensor self, Tensor? bias) -> (Tensor)
  self, bias: bwd_func(self, bias.has_value()? bias.value().scalar_type() : self.scalar_type())
```

This is very similar to what we have there for optional ScalarType.
```
- name: cumprod(Tensor self, int dim, *, ScalarType? dtype=None) -> Tensor
  self: cumprod_backward(grad.to(self.scalar_type()), self, dim, result)
  result: "cumprod_jvp(self_t, self_p, result, dim).to(dtype.has_value() ? *dtype : self_p.scalar_type())"
```

However, this doesn't work with optional tensor. In generated code, we seem to be treating `bias` inconsistently.
1. In `torch/csrc/autograd/generated/Functions.cpp`, it's taking bias as an at::Tensor , so it doesn't like `has_value` / `value`.
2. Meanwhile, `torch/csrc/autograd/generated/VariableType_2.cpp` is generating code for bias as `optional<at::Tensor>`.

So in the end I can't access any tensor wrapped as an optional.

## Additional context

The motivation is to update `native_layer_norm_backward`. Currently we are taking `bias` as an input here, but we are only using `bias.scalar_type()`.

https://github.com/pytorch/pytorch/blob/e6a3154519721eec323b227ea8dfbd05409cfe37/aten/src/ATen/native/native_functions.yaml#L2393

note that even though `bias` should have the same scalar_type as with `weight`, it is possible for `weight` to be `None` while `bias` is present, so we can't get rid of `bias`.

So we would like to update the API to take a ScalarType instead of optional tensor. Where we would need to update autograd to pass the scalar_type of an optional tensor

https://github.com/pytorch/pytorch/blob/e6a3154519721eec323b227ea8dfbd05409cfe37/tools/autograd/derivatives.yaml#L1013

**Comment by albanD** [comment]
I don't think we want to support optional Tensors here actually. Because that would still save the full Tensor. So you might as well pass `bias` as-is in this case.

We should teach our replacement pass [here](https://github.com/pytorch/pytorch/blob/c7db642a72d80facf9c8a17f42ac0bb04d04723f/tools/autograd/load_derivatives.py#L464) to recognize these patterns and properly save them as a plain scalar type.
